ext/opentelemetry-ext-wsgi/src/opentelemetry/ext/wsgi/__init__.py

```diff
@@ -88,14 +118,27 @@ def __call__(self, environ, start_response):
         tracer = trace.tracer()
         path_info = environ["PATH_INFO"] or "/"
 
-        with tracer.start_span(path_info, kind=trace.SpanKind.SERVER) as span:
+        span = tracer.create_span(path_info, kind=trace.SpanKind.SERVER)
+        span.start()
+        try:
             self._add_request_attributes(span, environ)
             start_response = self._create_start_response(span, start_response)
 
             iterable = self.wsgi(environ, start_response)
-            try:
-                for yielded in iterable:
-                    yield yielded
-            finally:
-                if hasattr(iterable, "close"):
-                    iterable.close()
+
+            # Put this in a subfunction to not delay the call to the wrapped
+            # WSGI application (instrumentation should change the application
+            # behavior as little as possible).
+            def iter_result():
+                try:
+                    for yielded in iterable:
+                        yield yielded
+                finally:
+                    if hasattr(iterable, "close"):
+                        iterable.close()
+                    span.end()
+
+            return iter_result()
+        except:  # noqa
+            span.end()
+            raise
```

Comment:
Why raise again here instead of making this a `finally` block?

Reply:
I guess it was because the assumption that `Span.end()` gets called in `iter_result` when there is no exception.
This piece of code is a bit risky and hard to maintain IMHO.

Reply:
If we decided to go with this approach, I'll be okay if we can put a comment like this :)
```python
# ATTENTION!!! HIGH VOLTAGE!!!
```

Reply:
If we can determine if a span has ended, then `finally: if not ended: end()` would be good.

Reply:
> If we decided to go with this approach

I'm all for a better approach if somebody finds one that still doesn't delay the app invocation.